### PR TITLE
[JENKINS-25509] Don't upload to S3 on aborted build

### PR DESCRIPTION
**Pull request by oxmane**
Signed-off-by: Eli Oxman <eli@alooma.io>

### CODE REVIEW

**Comment by king--code** [comment]
PR looks fine, I'd like to see it merged.

**Comment by ruZZil** [comment]
Any plan to integrate this feature soon? 
I'm going to compile the plugin with it for now..

**Comment by Jimilian** [comment]
Hi, @oxmane!

Are you still interesting in this PR? Can you solve conflicts? In this case I would be able to merge it.

Also, I think it's better to avoid any uploads for aborted jobs - with or without enabled 'noUploadOnFailure`. And it should be clear from logs do we skip uploading because job failed or because it was aborted.

**Comment by oxmane** [comment]
Hi @Jimilian !

Thanks for the feedback.
I've fixed the conflicts, and added some fixes following your feedback.
Please have a look at the PR again, and let me know if there's anything else to improve.

Thanks

File: src/main/java/hudson/plugins/s3/S3BucketPublisher.java [hunk 1]
```diff
@@ -141,8 +144,13 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath ws, @Nonnull Launc
             final List<FingerprintRecord> artifacts = Lists.newArrayList();
 
             for (Entry entry : entries) {
+                if (buildAborted) {
```
Comment (Jimilian) [inline src/main/java/hudson/plugins/s3/S3BucketPublisher.java]: If build is aborted, it's aborted for all entries. So, you can easily move this check on top of this method and remove `buildAborted` variable.

**Comment by oxmane** [comment]
@Jimilian did you have something like the current code in mind?

**Comment by Jimilian** [comment]
@oxmane, yes. Thanks for your PR, I will try to test and release new version on this week.

**Comment by badger451** [comment]
Any chance this could be an option in the plugin?  Personally, I want to archive if the build aborted.  My use case is if the build times out it shows up as aborted.  Having the archives will help figure out why the build timed out.

**Pull request merged**